js/validation/things.js

```diff
@@ -95,6 +95,10 @@ function validateThing(parentFieldName, typeFieldValidators, doc) {
                     // Was checked before
                     return true;
                 }
+                else if (fieldName.startsWith("codemeta:") || fieldName.startsWith("schema:")) {
+                    // Do not check fields from other versions FIXME ?
+                    return true;
+                }
```

Comment:
Can you remind me why this is needed?

Reply:
It could happen when validating of multi-version document, where, for instance, in this case:
```
    {
              "@context": "https://w3id.org/codemeta/3.0",
              "type": "SoftwareSourceCode",
              "name": "My Test Software",
              "continuousIntegration": "https://test-ci.org/my-software",
              "codemeta:contIntegration": {
                  "id": "https://test-ci.org/my-software"
              }
      }
```
`continuousIntegration` could be validated but not `codemeta:contIntegration`. But this hack is removed with https://github.com/hjonin/codemeta-generator/pull/1.

Reply:
Right. But then the "Unknown field" is never triggered because this addition allows every unknown property in the codemeta and schema.org namespaces.

Could you make this hack more specific, by explicitly listing the few properties that were added/removed? (and then remove it in the future of course)

Reply:
Done in the last commit.